### PR TITLE
add Slack error handling and remove use of deprecated API method

### LINKED ISSUE #111: Slack environment variable should not be "must set" variable
Slack notification is not "must have feature".
If the env var is not set, workflow should run properly and notification is not launched.

https://github.com/m3dev/gokart/blob/master/gokart/run.py#L32

### PR DESCRIPTION
**Pull request by hirosassa**
## What is this PR?

fixes #111 

I implemented following features:

1. add error handling and ignoring of `SlackApiError` and `ChannelNotFoundError`
2. removed deprecated API method: `channels.list` and simplify pagination handling [[reference link](https://api.slack.com/changelog/2020-01-deprecating-antecedents-to-the-conversations-api)]

### CODE REVIEW

**Comment by hirosassa** [comment]
@vaaaaanquish @Hi-king Plz review!

File: gokart/slack/slack_api.py [hunk 1]
```diff
@@ -24,33 +25,27 @@ def __init__(self, token, channel: str, to_user: str) -> None:
...
-            return channels
-
     def _get_channel_id(self, channel_name):
-        for channel in self._get_channels():
-            if channel['name'] == channel_name:
-                return channel['id']
-        raise ChannelNotFoundError(f'Channel {channel_name} is not found in public channels.')
+        params = {'exclude_archived': True, 'limit': 100}
+        try:
+            for channels in self._client.conversations_list(params=params):
```
Comment (hirosassa) [inline gokart/slack/slack_api.py:31]: Thanks to [SlackResponse](https://github.com/slackapi/python-slackclient/blob/main/slack/web/slack_response.py), it has __next__ methond, we can remove redundant lines of 
code for pagination handling.

File: gokart/slack/slack_api.py [hunk 2]
```diff
@@ -24,33 +25,27 @@ def __init__(self, token, channel: str, to_user: str) -> None:
...
+                if not channels:
+                    raise ChannelListNotLoadedError('Channel list is empty.')
+                for channel in channels.get('channels', []):
+                    if channel['name'] == channel_name:
+                        return channel['id']
+            raise ChannelNotFoundError(f'Channel {channel_name} is not found in public channels.')
+        except (ChannelNotFoundError, SlackApiError) as e:
+            logger.warning(f'The job will start without slack notification: {e}')
```
Comment (hirosassa) [inline gokart/slack/slack_api.py:39]: ignoring error, because slack notification is an nice-to-have feature

File: test/slack/test_slack_api.py [hunk 3]
```diff
@@ -3,65 +3,118 @@
...
         mock_client = MagicMock()
-        mock_client.api_call = MagicMock(side_effect=_channels_list)
+        mock_client.conversations_list = MagicMock(side_effect=_conversations_list)
         patch.return_value = mock_client
 
-        with self.assertRaises(gokart.slack.slack_api.ChannelListNotLoadedError):
+        with LogCapture() as l:
             gokart.slack.SlackAPI(token='invalid', channel='test', to_user='test user')
+            l.check(('gokart.slack.slack_api', 'WARNING',
+                     'The job will start without slack notification: Channel test is not found in public channels.'))
```
Comment (hirosassa) [inline test/slack/test_slack_api.py:38]: `LogCapture` enables testing log outputs

File: test/slack/test_slack_api.py [hunk 4]
```diff
@@ -3,65 +3,118 @@
...
+from testfixtures import LogCapture
+from slack.web.slack_response import SlackResponse
+from slack import WebClient
+
 import gokart.slack
 
 logger = getLogger(__name__)
 
 
+def _slack_response(token, data):
```
Comment (hirosassa) [inline test/slack/test_slack_api.py:15]: helper function for generating SlackResponse

**Comment by vaaaaanquish** [comment]
LGTM. Great unittests:)

File: gokart/slack/README.md [hunk 5]
```diff
@@ -0,0 +1,32 @@
+# goakrt Slack notification
```
Comment (Hi-king) [inline gokart/slack/README.md:1]:
```suggestion
# gokart Slack notification
```
😸

**Comment by Hi-king** [comment]
@hirosassa How about separating docs from code. My suggestion is moving gokart/slack/README.md to docs/slack.md.

**Review by Hi-king — CHANGES_REQUESTED** [review]
Need a little bit more discussion about where to put docs :)

File: gokart/slack/README.md (hunk 5 above)
Comment (hirosassa) [reply gokart/slack/README.md]: Oh! Thanks for your suggestion!

**Comment by Hi-king** [comment]
And I found that all docs in `docs` directory are rst formatted :)

**Comment by hirosassa** [comment]
> My suggestion is moving gokart/slack/README.md to docs/slack.md.
@Hi-king 
I agree with your idea. I rewrite the document on rst format!

**Review by Hi-king — APPROVED** [review]

**Comment by Hi-king** [comment]
@hirosassa thanks!

**Pull request merged**